### ISSUE TITLE
Filtering by clicking cell in table visualization doesn't wrap field in fk-> form when needed

### ISSUE THREAD

**Issue by MrMauricioLeite** [metabase/metabase#2738]
If this is a bug report, please fill in the blanks:

*  I am using the Chrome browser.
*  My computer's OS is Ubuntu.
*  I'm running Metabase Locally from the JAR.
*  My database is none, I'm using H2 Demo Database.
*  My Metabase version is TRUNK (just built it from github).

Well guys, I've just built a Metabase from source here to test some new feature since the 0.18.0 version is not out yet. I've just noticed that there is a bug with this feature we like to use a lot here, please take a look at the screenshots below:

![Basic question on test database](https://cloud.githubusercontent.com/assets/1953457/16015781/bc693b18-316d-11e6-9846-4396bd15bd4f.PNG)
![click any city name and click the equal sign](https://cloud.githubusercontent.com/assets/1953457/16015808/d3f9bc08-316d-11e6-8d70-3ad6a516234b.PNG)
![there is a new filter for the FIELD field, that obviously give an error](https://cloud.githubusercontent.com/assets/1953457/16015809/d3faebf0-316d-11e6-92ed-3b58f7717317.PNG)
![there goes the full error message](https://cloud.githubusercontent.com/assets/1953457/16015810/d3fe7496-316d-11e6-830c-be198f3829a3.PNG)

`Query expansion failed: could not find table 2 (FK ID = null). Resolved tables: {[nil 1] #metabase.models.table.TableInstance{:schema "PUBLIC", :name "ORDERS", :id 1}, [7 2] {:name "PEOPLE__via__USER_ID", :id 2}}`

I'm trying to find the problem on source to send a PR for fixing it. If anyone finds it first, please PR.

**Comment by camsaul** [comment]
Ok I figured out what's going on here. `user.city` is a field from another table but we're not wrapping it in the `fk->` form in this case.

This is the query we're producing when you click the value:
```clojure
{:source_table 451
 :aggregation  ["sum" 1206]
 :breakout     [["fk->" 1207 1210]]
 :filter       ["AND" ["TIME_INTERVAL" 1201 "last" "month"]
                      ["=" 1210 "Abbeyhaven"]]              ; missing fk-> form
 :order_by     [[["fk->" 1207 1210] "ascending"]]}
```

Instead it should be
```clojure
{:source_table 451
 :aggregation  ["sum" 1206]
 :breakout     [["fk->" 1207 1210]]
 :filter       ["AND" ["TIME_INTERVAL" 1201 "last" "month"]
                      ["=" ["fk->" 1207 1210] "Abbeyhaven"]]
 :order_by     [[["fk->" 1207 1210] "ascending"]]}
```

**Comment by camsaul** [comment]
I have a fix for this I think

**Comment by tlrobinson** [comment]
I think this is related to @salsakran's desire to better abstract "dimensions".

I'm not sure if it's exactly what @salsakran has in mind, but a simple fix could be to return a `dimension` (or whatever name we decide is appropriate) property on the `City` column in the dataset result, which includes the full fk clause, and we'd use that instead of `id`, so that the visualization code doesn't need to be aware of the fact that the column is being referred to through a fk.

Related: We need to be more careful about which operators we show in the "quick filter" options. You can incorrectly apply a ">" or "<" to text columns. I thought there was a ticket open for that but I can't find it.

**Comment by MrMauricioLeite** [comment]
Great @camsaul, congrats finding the problem. If you have any fix, please let me take a look, I really want to contribute with the code and would be great to learn a bit more about Metabase source.

I agree with you @tlrobinson, we talked to our users so they should not use a > or < filter on text columns cause that makes no sense but be sure, they tried. Disabling the buttons would be a good start, or not showing at all.

**Comment by agilliland** [comment]
The code for handling those quick filters is a redux action which takes in `(rowIndex, columnIndex, filterOperation)` and then expects the appropriate changes to the query to be handled from there ... https://github.com/metabase/metabase/blob/master/frontend/src/metabase/query_builder/actions.js#L635-L700

When the quick filter code was originally put it it was just a prototype that was merged in without any real hardening, so most likely this has been broken for quite a while.  And as Tom said, there are other issues with it such as not properly aligning the filter operations based on the type of data in the cell.

In any case, the operation is at least well abstracted from the visualization code because all that is triggered from the Table visualization is this callback indicating that the user tried to filter a given cell on the result, and the visualization code has no idea what was actually in that cell.  The solution would be to tweak that function to properly handle the situation where the column that was being filtered was an FK.

**Comment by MrMauricioLeite** [comment]
This is surely not broken for a while, just on master branch. 
I'm using it on 0.17.1 and we have no problem using this feature. 

Give it a try if needed.

**Comment by tlrobinson** [comment]
Interesting, it does indeed work on v0.17.1, but it's sending the same query without the `->fk` clause, so it seems like something about the query processor changed. That said, I'm surprised it used to work at all without the `fk->` clause, and would still consider this a frontend bug.

**Comment by camsaul** [comment]
Yeah when I did the food to handle multiple joins against the same table I had to tighten up the logic that figures out how to wire up columns from joined tables. So it's a little stricter now than it was

**Comment by camsaul** [comment]
Fixed by #2752
